rs/sdk/src/api/command/command_handler.rs

```diff
@@ -55,8 +55,10 @@ impl<R> CommandHandlerRegistry<R> {
         jwt: &str,
         public_key: &str,
         now: TimestampMillis,
+        shared_state: S,
```

Comment:
If we have to pass in the shared state every time that would look messy for the majority of use cases where the shared state will be `()`.
So maybe we instead hold the shared state in the `CommandHandlerRegistry`?
Then it automatically passes it through to `command_handler.execute`.
Or are you envisioning potentially passing in different instances, but the type always remains the same?